packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -478,3 +541,50 @@ function createSignatureCache(): InMemoryLRUCache<string> {
 export function signatureCacheKey(queryHash: string, operationName: string) {
   return `${queryHash}${operationName && ':' + operationName}`;
 }
+
+// Helper function to modify the EngineReportingOptions if the deprecated fields 'privateVariables' and 'privateHeaders'
+// were set.
+// - Throws an error if both the deprecated option and its replacement (e.g. 'privateVariables' and 'sendVariableValues') were set.
+// - Otherwise, wraps the deprecated option into objects that can be passed to the new replacement field (see the helper
+// function makeSendValuesBaseOptionsFromLegacy), and deletes the deprecated field from the options
+export function handleLegacyOptions(
+  options: EngineReportingOptions<any>,
+): void {
+  // Handle the legacy option: privateVariables
+  if (options.privateVariables != null && options.sendVariableValues) {
```

Comment:
While I agree that strict inequality is more clear, in this case `undefined` is more likely to be the thing we're comparing against, right?

Reply:
I think you're right, @glasser.

Perhaps `!options.privateVariables` is more appropriate then?

This notation seems it might lead someone to believe that it's checking for "not null", which is kinda my gut when reading it since in other cases explicit `null` has semantic meaning.

Reply:
But then that ignores `privateVariables: false` (which used to be the default but isn't any more)...

Honestly `!== undefined` is maybe ok — null is not actually documented to be a valid value...

Reply:
@glasser 	I'm okay with that!  Though a preference for `typeof v !== 'undefined'` to avoid ye' ole JavaScript gotchaz:

```
(function () {
  var undefined = true; console.log(undefined);
})(); // prints true
```

(`undefined` is still not a reserved word, even if the global `undefined` is no longer as mutable as it once was!)

Reply:
I didn't use strict equality here because I wanted to make sure the `null` input was caught as well. I guess this wouldn't be a valid TS input, but I also have checks elsewhere for invalid TS inputs (i.e. `{none: false}`, as opposed to `{none: true}`)..

But maybe for clarity, I should explicitly write out the cases!

Reply:
re typeof: uh seriously??? I kind of feel like if you redefine global `undefined` you get to keep both pieces.

I have always understood that the `typeof foo === 'undefined'` pattern should be reserved for the specific case where `foo` is a literal that is not a variable defined explicitly in that scope and you want to know if it exists as a global or not (often in generated code, or code trying to deal with old libraries that export via globals, or probing the JS implementation for globals).